hedera-node/src/main/java/com/hedera/services/state/expiry/ExpiryManager.java

```diff
@@ -54,6 +55,13 @@
  * </ol>
  */
 public class ExpiryManager {
+	/* Since the key in Pair<Long, Consumer<EntityId>> is the schedule entity number---and
+	entity numbers are unique---the downstream comparator below will guarantee a fixed
+	ordering for ExpiryEvents with the same expiry. */
+	private static final Comparator<ExpiryEvent<Pair<Long, Consumer<EntityId>>>> PQ_CMP = Comparator
+			.comparingLong(ExpiryEvent<Pair<Long, Consumer<EntityId>>>::getExpiry)
+			.thenComparingLong(ee -> ee.getId().getKey());
```

Comment:
So before this change, we compared only expiry as the removed code in `ExpiryEvent`, and now we compare expiry first then the schedule entity number? Why can we have the same expiry for different scheduled entities?

Reply:
This can happen because we round expiration times to a consensus second. 

So if two txns are scheduled in the same consensus second `t`, both will expire at consensus second `t + 1800`.

Reply:
Agreed. Thanks for the explanation.

Reply:
Then perhaps we should add this explanation in the code here.

Reply:
Added this explanation.